org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -304,4 +310,43 @@ public ClasspathResult(URI projectRoot, String[] classpaths, String[] modulepath
 			this.modulepaths = modulepaths;
 		}
 	}
+
+	public static SymbolInformation resolve(SymbolInformation request) {
+		ITypeRoot unit = JDTUtils.resolveTypeRoot(request.getLocation().getUri());
+		if (unit == null || !unit.exists()) {
+			return null;
+		}
+		Location location = null;
+		IJavaElement element = null;
+
+		try {
+			Deque<IJavaElement> children = new ArrayDeque<>(Arrays.asList(unit.getChildren()));
+			while (element == null && !children.isEmpty()) {
+				IJavaElement child = children.pop();
+				if (child instanceof IType) {
+
+					if (request.getName().equals(child.getElementName())) {
+						element = child;
+						break;
+					}
+					children.addAll(Arrays.asList(((IParent) child).getChildren()));
+				}
+
+			}
+			location = JDTUtils.toLocation(element);
+
+			SymbolInformation si = new SymbolInformation();
+			si.setName(request.getName());
+			si.setContainerName(request.getContainerName());
+			si.setLocation(location);
+			si.setKind(request.getKind());
+
+			return si;
+
+		} catch (JavaModelException e) {
+			JavaLanguageServerPlugin.logError("Problem resolving symbol information for " + unit.getElementName());
+		}
+
+		return null;
```

Comment:
I would return the original request back for the failed case. Others look good to me.